array/array.go

```diff
@@ -223,56 +223,146 @@ func (a *String) ValueLen(i int) int {
 }
 
 type StringBuilder struct {
-	b *array.BinaryBuilder
+	b            *array.BinaryBuilder
+	mem          memory.Allocator
+	value        string
+	length       int
+	capacity     int
+	dataCapacity int
+	refCount     int
 }
 
 func NewStringBuilder(mem memory.Allocator) *StringBuilder {
 	return &StringBuilder{
-		b: array.NewBinaryBuilder(mem, arrow.BinaryTypes.String),
+		mem:      mem,
+		refCount: 1,
+	}
+}
+func (b *StringBuilder) init() {
+	if b.b == nil {
```

Comment:
Yeah, this is what I mean: `b` is not a great name, but it's kinda idiomatic to go to same your "self" variable like that. `b.b` on the other hand? I have no idea what that is.

I'm not sure there's anything actionable now, because team inertia, etc, but I think it'd be a net benefit for our team if we started thinking about whether we want to move away from this.

Reply:
I can just change it to builder. I don't really have a problem with that.